src/runrms/api/worker.py

```diff
@@ -0,0 +1,310 @@
+        """Handle shutdown signals."""
+        self.running = False
+
+    def setup_zmq(self) -> None:
+        """Setup ZMQ context and socket."""
+        self._context = zmq.Context()
+        self._socket = self.context.socket(zmq.REP)
+        self._socket.bind(self.zmq_address)
+        self._socket.setsockopt(zmq.RCVTIMEO, 1000)  # 1 sec timeout
+
+    @property
+    def context(self) -> zmq.Context[zmq.Socket[bytes]]:
+        if self._context is None:
+            raise RuntimeError("ZMQ context not established.")
+        return self._context
+
+    @property
+    def socket(self) -> zmq.Socket[bytes]:
+        if self._socket is None:
+            raise RuntimeError("ZMQ socket not established.")
+        return self._socket
+
+    def _resolve_object(self, request: Request) -> tuple[Any, list[str]]:
+        """Resolve object from request path.
+
+        Start at the 'root' of 'rmsapi', which is 'rmsapi' itself.
+        In the following logic we will drill down to the actual object we need to
+        work on, which may be something new, specified by a 'path', or it may
+        actually be a reference to an existing object we've stored in the object
+        store, and which was sent with the request (a ProxyRef dict).
+        """
+        obj: Any = self.api_object
+        path_to_traverse = request.path
+
+        # This is a reference to an object we've seen and stored before.
+        if request.path and request.path[0] == "$ref":
+            # A reference was sent, but not its id!
+            if len(request.path) < 2:
+                raise ValueError("$ref requires an object id")
+
+            obj_id = request.path[1]
+            if obj_id not in self.object_store:
+                raise KeyError(f"Object id {obj_id} not found in store")
+
+            obj = self.object_store[obj_id]
+            # We have the reference object. But are we accessing things on it?
+            # We still have to traverse the path to see, and it depends on the RPC
+            # call, too.
+            path_to_traverse = request.path[2:]
+
+        return obj, path_to_traverse
+
+    def execute_request(self, request: Request) -> Response:
+        """Execute a request and return response."""
+        try:
+            if request.msg_type == "ping":
+                return Response(success=True, value="pong")
+
+            if request.msg_type == "shutdown":
+                self.running = False
+                return Response(success=True, value="shutting down")
+
+            obj, path = self._resolve_object(request)
+
+            if request.msg_type == "getattr":
+                result = self._execute_getattr(obj, path)
+            elif request.msg_type == "call":
+                result = self._execute_call(obj, path, request.args, request.kwargs)
+            elif request.msg_type == "setattr":
+                self._execute_setattr(obj, path, request.value)
```

Comment:
no `result` defined here, but this method always `return self._make_response(result)`, so i think setattr call will always fail. maybe do something like this


```suggestion
                self._execute_setattr(obj, path, request.value)
                result = None
```